src/Compilers/Core/Portable/Diagnostic/CommonDiagnosticComparer.cs

```diff
@@ -25,7 +25,7 @@ public bool Equals(Diagnostic x, Diagnostic y)
                 return false;
             }
 
-            return x.Location == y.Location && x.Id == y.Id;
+            return x is { Location: y.Location, Id: y.Id };
```

Comment:
>y.Location [](start = 36, length = 10)

Not a valid pattern.

Reply:
➡️ Treating this one as a duplicate of https://github.com/dotnet/roslyn/pull/39738#discussion_r344416577